datahub-web-react/src/app/ingest/source/builder/RecipeForm/common.tsx

```diff
@@ -55,13 +56,11 @@ function clearFieldAndParents(recipe: any, fieldPath: string | string[]) {
 }
 export function setFieldValueOnRecipe(recipe: any, value: any, fieldPath: string | string[]) {
     const updatedRecipe = { ...recipe };
-    if (value !== undefined) {
-        if (value === null || value === '') {
-            clearFieldAndParents(updatedRecipe, fieldPath);
-            return updatedRecipe;
-        }
-        set(updatedRecipe, fieldPath, value);
+    if (value === null || value === '' || value === undefined) {
```

Comment:
did you find out when `undefined` gets passed in here?

Reply:
like I said before, I'm pretty sure this check was when we would pass extra values in here that shouldn't be updated but has since been refactored, so I'm pretty sure this is a safe change to make (also if it's undefined it makes sense we should clear it and its parents if they're empty as well)

Reply:
I have not found any issues while testing it out locally...